src/main/java/htsjdk/samtools/CRAMIterator.java

```diff
@@ -149,11 +149,11 @@ void nextContainer() throws IOException, IllegalArgumentException,
 
         records.clear();
         if (cramRecords == null)
-            cramRecords = new ArrayList<CramCompressionRecord>(container.nofRecords);
+            cramRecords = new ArrayList<>(container.nofRecords);
         else
             cramRecords.clear();
 
-        parser.getRecords(container, cramRecords, validationStringency);
+        parser.getRecords(cramRecords, container, validationStringency);
```

Comment:
Reads records into `cramRecords` which is currently a class member.  We could change this to create a new `cramRecords` List here instead, but I left it as-is because I assume there's some reason for it.